File: src/types/product.ts
// ATENÇÃO: este arquivo continha, por engano, o código do hook useFetch.
// Corrigido para conter os tipos de domínio de produto (pizzas), como previsto no plano.

export type Size = "P" | "M" | "G";

export interface Topping {
  id: string;
  name: string;
  price: number;
}

export interface Pizza {
  id: string;
  name: string;
  description: string;
  imageUrl: string;
  basePrice: number;
  category: string;
  availableSizes: Size[];
  availableToppings: Topping[];
}

// Payload usado pelo admin ao criar/editar uma pizza (sem id, gerado pela API mock)
export type PizzaInput = Omit<Pizza, "id">;

File: backend/app/schemas/product.py
from app.schemas.common import CamelModel


class ToppingOut(CamelModel):
    id: str
    name: str
    price: float


class ToppingInput(CamelModel):
    """Usado dentro de ProductInput — sem id (gerado pela API ao salvar)."""

    name: str
    price: float


class ProductBase(CamelModel):
    name: str
    description: str = ""
    image_url: str = ""
    category: str
    base_price: float
    available_sizes: list[str] = []  # subset de ["P", "M", "G"]


class ProductInput(ProductBase):
    """Payload de POST/PUT /products."""

    available_toppings: list[ToppingInput] = []


class ProductOut(ProductBase):
    id: str
    available_toppings: list[ToppingOut] = []
